recipes/full_finetune_single_device.py

```diff
@@ -430,7 +435,7 @@ def _setup_model(
 
         log.info(f"Model is initialized with precision {self._dtype}.")
 
-        if self._device.type == "cuda":
+        if self._device.type in DeviceSupport.get_cuda_like_device_types():
```

Comment:
Maybe it's just me but I find this to be a bit confusing.. can we just scrap this method and use `if self._device.type != "cpu"` in these places instead? I know it may not be as general but I think this is extra indirection that we don't really need

Reply:
I think you’re right! This encapsulation seems redundant at the moment, as cuda-like devices currently appear to be simply non-CPU devices.